apps/web/src/scenes/TokenDetailPage/TokenDetailAsset.tsx

```diff
@@ -85,6 +92,17 @@ function TokenDetailAsset({ tokenRef, hasExtraPaddingForNote }: Props) {
     handleNftLoaded();
   }, [imageUrl, handleNftLoaded, cacheLoadedImageUrls, tokenId, resultDimensions]);
 
+  const [showLightbox, setShowLightbox] = useState(false);
+  const toggleLightbox = useCallback(() => {
+    setShowLightbox((prev) => !prev);
+  }, []);
+  const [lightboxContainer, setLightboxContainer] = useState<HTMLElement | null>(null);
+  useEffect(() => {
+    setLightboxContainer(document.getElementById(`lightbox-portal-${token.dbid}`));
```

Comment:
nit: could we have a util for generating the lightbox portal element ID with token ID:
```ts
function getLightboxPortalElementId(tokenId: string) {
  return `lightbox-portal-${tokenId}`
}
```

Reply:
nit: for these functions from L95-104, should we encapsulate it into a single `useLightbox` hook? just so we don't miss anything if we try to set it up elsewhere

```ts
function useLightbox(tokenId: string) {
  const [showLightbox, setShowLightbox] = useState(false);
  const toggleLightbox = useCallback(() => {
    setShowLightbox((prev) => !prev);
  }, []);
  const [lightboxContainer, setLightboxContainer] = useState<HTMLElement | null>(null);
  useEffect(() => {
    setLightboxContainer(document.getElementById(`lightbox-portal-${tokenId}`));
    // only need to run on mount
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return { showLightbox, toggleLightbox, lightboxContainer }
}
```

Reply:
good idea! added `getLightboxPortalElementId`

for the hook, I ended up separating some of the logic so I think it makes less sense now to consolidate into a hook, so left as is. lmk if it still looks like we might benefit from it